src/components/hub/elements/MessagesList.js

```diff
@@ -56,74 +57,85 @@ export default function MessagesList({ scrollRef }) {
       return;
     }
 
-    api
-      .messageList({
-        cid: selectedCID,
-        limit: +process.env.REACT_APP_MESSAGES_COUNT_TO_PRELOAD,
-        updated_at: { lt: messages[0].created_at },
-      })
-      .then((arr) => {
-        if (!arr.length) {
-          needToGetMoreMessage.current = false;
-          return;
+    const processMessages = (arr) => {
+      indexedDB.insertManyMessages(arr);
+
+      if (!arr.length) {
+        needToGetMoreMessage.current = false;
+        return;
+      }
+
+      const messagesIds = arr.map((el) => el._id).reverse();
+      needToGetMoreMessage.current = !(
+        messagesIds.length < +process.env.REACT_APP_MESSAGES_COUNT_TO_PRELOAD
+      );
+
+      updateParticipantsFromMessages(arr);
+      dispatch(addMessages(arr));
+      dispatch(
+        upsertChat({
+          _id: selectedCID,
+          messagesIds: [
+            ...new Set([...messagesIds, ...messages.map((el) => el._id)]),
+          ],
+          activated: true,
+        })
+      );
+      const mAttachments = {};
```

Comment:
To be honest, this code looks very similar to `messageService.syncData`. Can we reuse/optimise?